include/tvm/runtime/crt/aot/tvm_backend.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file include/tvm/runtime/crt/aot/tvm_backend.h
+ * \brief Backend functions for the AOT executor
+ *
+ * These are not designed to user-facing and may change without warning
+ */
+
+#ifndef TVM_RUNTIME_CRT_AOT_TVM_BACKEND_H_
```

Comment:
I think we should try and generally reduce "ways to do something" in the code base because we already have many bifurcated mechanisms. If an interface isn't working we should redesign to encompass new use cases.

Reply:
So, this was a controversial subject for us. The main issue we had with reusing the existing `crt` was that we are aiming to have a more friendly code generation process targeting embedded processors. In particular, we have issues with:
* TVMBackendMemoryAllocWorkspace to take an uint64_t to specify the size
* TVMValue to be a 64 bit datastructure
* Dependency on DLTensor and DLDevice
However, TVMValue and DLTensor will be removed with the next few patches and TVMBackendMemoryAllocWorkspace will be the subject of global memory unification. So we refactored the code to fit in `crt` but I wanted just to flag that this brings a lot of baggage that for embedded devices is not essential